docs/source/deployment/lambda.mdx

```diff
@@ -284,42 +252,177 @@ If you ever want to remove the S3 bucket or Lambda functions that Serverless cre
  serverless remove
 ```
 
-## Customizing HTTP behavior
+## Middleware
 
-The `@as-integrations/aws-lambda` package is compatible with the following event types `APIGatewayProxyEvent`, `APIGatewayProxyEventV2`, `ALBEvent`. This supports a wide range of services like API Gateway HTTP Proxy APIs, API Gateway REST Proxy APIs, Lambda Function URLs, and Application Load Balancers. However, it does not let you customize HTTP behavior directly or support other AWS products that invoke Lambda functions (e.g., S3 or DynamoDB).
+In order to implement event and result mutations, type-safe middleware can be passed to the `startServerAndCreateLambdaHandler` call. The API is as follows:
 
-If you want to customize your HTTP behavior, you can couple Apollo Server's Express integration (i.e., [`expressMiddleware`](../api/express-middleware)) with the [`@vendia/serverless-express`](https://github.com/vendia/serverless-express) package. The `@vendia/serverless-express` library translates between Lambda events and Express requests. Despite their similar names, the Serverless CLI and the `@vendia/serverless-express` package are unrelated.
+```ts
 
-You can update your Apollo Server setup to the following to have a fully functioning Lambda server that works in a variety of AWS features:
+import { middleware, startServerAndCreateLambdaHandler, handlers } from "@as-integrations/aws-lambda";
 
-<MultiCodeBlock>
+const requestHandler = handlers.createAPIGatewayProxyEventV2RequestHandler()
+
+// All middleware must be async and will be strictly typed based on the `requestHandler`'s types
+const middlewareFn: middleware.MiddlewareFn<typeof requestHandler> = async (event) => {
+  // event is intended to be mutable and can be updated here
+
+  // optional result mutation, otherwise function returns `Promise<void>`
+  return async (result) => {
+    // result is intended to be mutable and can be updated here
+
+    // `Promise<void>` return
+  }
+}
```

Comment:
I don't have a great suggestion here besides maybe just "don't say anything", but suggesting returning `Promise<void>` is a little weird and possibly confusing when they can return nothing or implicitly return, though I agree that it's technically correct.

Suggesting some changes, but feel free to tweak them.
```suggestion
// Middleware is an async function whose type is based on your request handler. Middleware
// can read and mutate the incoming event. Additionally, returning an async function from your
// middleware allows you to read and mutate the result before it's sent.
const middlewareFn: middleware.MiddlewareFn<typeof requestHandler> = async (event) => {
  // read or update the event here

  // optionally return a callback to access the result
  return async (result) => {
    // read or update the result here
  }
}
```